pandas/tests/groupby/test_categorical.py

```diff
@@ -207,7 +209,8 @@ def test_level_get_group(observed):
     assert_frame_equal(result, expected)
 
 
-# GH#21636 previously flaky on py37
+# GH#21636 flaky on py37 is not solved, remove xfail when it is solved
+@pytest.mark.xfail(PY37, reason="flaky on 3.7, xref gh-21636", strict=False)
```

Comment:
should also condition on windows?  possibly even the numpy version?  we should be maximally restrictive here